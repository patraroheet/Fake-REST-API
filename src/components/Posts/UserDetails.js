import React from "react";
import { Card, CardHeader, CardBody } from "reactstrap";

export default function UserDetails({ user }) {
  return user ? (
    <Card>
      <CardHeader>{user.name}</CardHeader>
      <CardBody>
        <p id={"email"}>Email : {user.email}</p>
        <p id={"phone"}>Phone : {user.phone}</p>
        <address>
          {`Address : ${user.address.street}, ${user.address.suite}, ${user.address.city} - ${user.address.zipcode}`}
        </address>
      </CardBody>
    </Card>
  ) : null;
}
