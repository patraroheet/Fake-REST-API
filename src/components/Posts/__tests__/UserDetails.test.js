import React from "react";
import TestRenderer from "react-test-renderer";
import { mount } from "enzyme";
import UserDetails from "../UserDetails";

describe("User Details Component Testing", () => {
  let dummyUser = null;
  beforeEach(() => {
    dummyUser = {
      id: 1,
      name: "Leanne Graham",
      username: "Bret",
      email: "Sincere@april.biz",
      address: {
        street: "Kulas Light",
        suite: "Apt. 556",
        city: "Gwenborough",
        zipcode: "92998-3874",
        geo: {
          lat: "-37.3159",
          lng: "81.1496"
        }
      },
      phone: "1-770-736-8031 x56442",
      website: "hildegard.org",
      company: {
        name: "Romaguera-Crona",
        catchPhrase: "Multi-layered client-server neural-net",
        bs: "harness real-time e-markets"
      }
    };
  });
  it("snapshot testing", () => {
    const component = TestRenderer.create(<UserDetails user={dummyUser} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("shall show user details properly", () => {
    const wrapper = mount(<UserDetails />);
    wrapper.setProps({ user: dummyUser });
    // expect(wrapper.props.user).toBe({ ...dummyUser });
    const header = wrapper.find(".card-header");
    expect(header.text()).toBe(dummyUser.name);
    const email = wrapper.find("#email");
    expect(email.text()).toBe(`Email : ${dummyUser.email}`);
    const phone = wrapper.find("#phone");
    expect(phone.text()).toBe(`Phone : ${dummyUser.phone}`);
  });
});
