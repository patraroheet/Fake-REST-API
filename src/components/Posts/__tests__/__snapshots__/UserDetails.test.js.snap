// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Details Component Testing snapshot testing 1`] = `
<div
  className="card"
>
  <div
    className="card-header"
  >
    Leanne Graham
  </div>
  <div
    className="card-body"
  >
    <p
      id="email"
    >
      Email : 
      Sincere@april.biz
    </p>
    <p
      id="phone"
    >
      Phone : 
      1-770-736-8031 x56442
    </p>
    <address>
      Address : Kulas Light, Apt. 556, Gwenborough - 92998-3874
    </address>
  </div>
</div>
`;
